src/linear_algebra/basic.lean

```diff
@@ -400,6 +400,14 @@ def dom_restrict'
 @[simp] lemma dom_restrict'_apply (f : M →ₗ[R] M₂) (p : submodule R M) (x : p) :
   dom_restrict' p f x = f x := rfl
 
+variables (S : Type*) [monoid S] [distrib_mul_action S M₂] [smul_comm_class R S M₂]
+  [mul_action S R] [is_scalar_tower S R M₂]
+
+-- example application of this instance: if S and R are comm_semirings and R is an S-algebra
+-- then the S-actions and R-actions on Hom_R(M,M₂) coincide.
+instance : is_scalar_tower S R (M →ₗ[R] M₂) :=
```

Comment:
I think this generalizes to `is_scalar_tower S S' (M →ₗ[R]  N)`, with the right assumptions on `S'`

Reply:
This is the instance I'm thinking of:
```
instance {T : Type*} [monoid T] [has_scalar S T] [distrib_mul_action T M₂] [smul_comm_class R T M₂]
  [is_scalar_tower S T M₂] :
  is_scalar_tower S T (M →ₗ[R] M₂) :=
{ smul_assoc := λ _ _ _, ext $ λ _, smul_assoc _ _ _ }
```
which goes right under `smul_comm_class` and doesn't require `comm_ring R`

Reply:
I've now confirmed that this works in my use case. Thanks for figuring this out! It had taken me long enough to figure out my version, but I agree yours is even better (it's more general).